### PR TITLE
Add Logger For OpenTelemetry

### PR DESCRIPTION
**Pull request by emmayspark**
Logger will be passed by VmAgent. This is the first step to enable OpenTelemetry.

### CODE REVIEW

File: VmAgent.Core/MultiLogger.cs [hunk 1]
```diff
@@ -13,62 +13,74 @@ namespace Microsoft.Azure.Gaming.VmAgent.Core
     public class MultiLogger : ILogger
     {
         private readonly ILogger _logger;
-        private readonly TelemetryClient _genevaTelemetryClient;
+		private readonly TelemetryClient _genevaTelemetryClient;
+		private ILogger _genevaOTelLogger;
+        private const string VmAgentLoggerName = "VmAgentLogger";
 
-        public MultiLogger(ILogger logger, TelemetryClient genevaTelemetryClient)
+		public MultiLogger(ILogger logger, TelemetryClient genevaTelemetryClient, ILogger genevaLogger = null)
```
Comment (pedrofv) [inline VmAgent.Core/MultiLogger.cs:20]: >MultiLogger [](http://example.com/codeflow?start=9&length=11)

We should add comments on why we have 2 ILogger parameters here.

Are we going to remove/replace the existing ILogger in the future?

**Review by pedrofv — APPROVED** [review]
:shipit:

File: VmAgent.Core/MultiLogger.cs [hunk 2]
```diff
@@ -13,62 +13,74 @@ namespace Microsoft.Azure.Gaming.VmAgent.Core
...
-        }
+			_genevaTelemetryClient.TrackTrace(message, SeverityLevel.Error);
+			_genevaOTelLogger?.LogError(message);
+		}
 
         public void LogException(Exception exception)
         {
             string message = exception.ToString();
             _logger.LogError(message);
-            _genevaTelemetryClient.TrackTrace(message, SeverityLevel.Error);
```
Comment (yegarci) [inline VmAgent.Core/MultiLogger.cs:51]: Why is this _genevaTelemetryClient deleted, but the rest stayed?

**Review by yegarci — CHANGES_REQUESTED** [review]

**Review by yegarci — APPROVED** [review]

File: VmAgent.Core/MultiLogger.cs [hunk 3]
```diff
@@ -14,11 +14,22 @@ public class MultiLogger : ILogger
     {
         private readonly ILogger _logger;
         private readonly TelemetryClient _genevaTelemetryClient;
+        private ILogger _genevaOTelLogger;
```
Comment (dgkanatsios) [inline VmAgent.Core/MultiLogger.cs:17]: nit: _genevaOtelLogger (lowercase T?)

File: VmAgent.Core/MultiLogger.cs [hunk 4]
```diff
@@ -14,11 +14,22 @@ public class MultiLogger : ILogger
...
         {
             _logger = logger ?? throw new ArgumentException("logger cannot be null");
+
+            // To enable both GenevaAgentChannel and OpenTelemetry, we have 2 ILogger parameters here.
+            // We will replace GenevaAgentchannel with Opentelemetry, once we verify all logs both in test and prod environments.
             _genevaTelemetryClient = genevaTelemetryClient ?? throw new ArgumentException("genevaTelemetryClient cannot be null");
+            _genevaOTelLogger = genevaLogger;
+        }
+
+        public void SetUpLogger(ILoggerFactory loggerFactory)
```
Comment (dgkanatsios) [inline VmAgent.Core/MultiLogger.cs:30]: why do we have the separate method if we're setting the same variable on the constructor?

File: VmAgent.Core/MultiLogger.cs [hunk 5]
```diff
@@ -14,11 +14,16 @@ public class MultiLogger : ILogger
...
 
-        public MultiLogger(ILogger logger, TelemetryClient genevaTelemetryClient)
+        public MultiLogger(ILogger logger, TelemetryClient genevaTelemetryClient, ILogger genevaOpenTelemetryLogger = null)
         {
             _logger = logger ?? throw new ArgumentException("logger cannot be null");
+
+            // We have 2 ILogger parameters (1. Logger to write logs to local file. 2.Geneva OpenTelemetry Logger)
+            // We will replace GenevaAgentchannel with OpenTelemetry, once we verify all logs both in test and prod environments.
             _genevaTelemetryClient = genevaTelemetryClient ?? throw new ArgumentException("genevaTelemetryClient cannot be null");
+            _genevaOtelLogger = genevaOpenTelemetryLogger;
```
Comment (dgkanatsios) [inline VmAgent.Core/MultiLogger.cs:26]: what are the cases that this is null?

Comment (emmayspark) [reply VmAgent.Core/MultiLogger.cs:26]: it will be null in prod until we enable Otel in production.

**Pull request merged**